charts/influxdb/templates/_helpers.tpl

```diff
@@ -61,3 +61,22 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Format service config key values
+Numeric values require special handling due to the Helm upstream issue, otherwise they may be converted to scientific
+notation, for example 1073741824 -> 1.073741824e+09.
+Reference: https://github.com/helm/helm/issues/1707
+*/}}
+{{- define "influxdb.formatConfigMap" }}
+{{- range $key, $value := index . }}
+{{- $tp := typeOf $value }}
+{{- if eq $tp "string" }}
+{{ $key }} = {{ $value | quote }}
+{{- else if eq $tp "bool" }}
+{{ $key }} = {{ $value }}
+{{- else }}
+{{ $key }} = {{ $value | int64 }}
+{{- end  }}
+{{- end }}
+{{- end -}}
```

Comment:
```suggestion
{{- if eq $tp "string" }}
{{ $key }} = {{ $value | quote }}
{{- else }}
{{ $key }} = {{ $value }}
{{- end }}
{{- end }}
```

Factoring out is great! But let's address this logical change in a separate PR as there are concerns over the arrays, floats as mentioned in https://github.com/influxdata/helm-charts/pull/254#issuecomment-787414888

OOC, is there a value in your current influxdb config that is turning into scientific notation today @szymonpk ?

Reply:
```
    data:
        query-log-enabled: false
        cache-max-memory-size: 1073741824
```

IIRC, influxv1 does not accept quoted value here.